src/main/scala/dotty/dokka/ScalaContentBuilder.scala

```diff
@@ -0,0 +1,504 @@
+package dotty.dokka
+
+import org.jetbrains.dokka.base.translators.documentables.{DefaultPageCreator, PageContentBuilder, PageContentBuilder$DocumentableContentBuilder}
+import org.jetbrains.dokka.base.signatures.SignatureProvider
+import org.jetbrains.dokka.base.transformers.pages.comments.CommentsToContentConverter
+import org.jetbrains.dokka.transformers.documentation.DocumentableToPageTranslator
+import org.jetbrains.dokka.utilities.DokkaLogger
+import org.jetbrains.dokka.model._
+import org.jetbrains.dokka.pages._
+import collection.JavaConverters._
+import org.jetbrains.dokka.model.properties._
+import org.jetbrains.dokka.base.transformers.documentables.CallableExtensions
+import org.jetbrains.dokka.DokkaConfiguration$DokkaSourceSet
+import org.jetbrains.dokka.base.resolvers.anchors._
+import org.jetbrains.dokka.links._
+import org.jetbrains.dokka.model.properties.PropertyContainer
+import org.jetbrains.dokka.model.doc._
+
+class ScalaPageContentBuilder(
+    val commentsConverter: CommentsToContentConverter,
+    val signatureProvider: SignatureProvider,
+    val logger: DokkaLogger
+) { self =>
+    def contentForDRI(
+        dri: DRI,
+        sourceSets: Set[DokkaConfiguration$DokkaSourceSet],
+        kind: Kind = ContentKind.Main,
+        styles: Set[Style] = Set(),
+        extra: PropertyContainer[ContentNode] = PropertyContainer.Companion.empty(),
+        buildBlock: ScalaPageContentBuilder#ScalaDocumentableContentBuilder => ScalaPageContentBuilder#ScalaDocumentableContentBuilder
```

Comment:
Is it necessary to use the `ScalaPageContentBuilder#` prefix? It shouldn't, since we're inside that class; if it is, it's a bug in Dotty.

Reply:
I've checked it right now and actually it's not needed, but I had some sort of habit after writing code like that in other class, where it was necessary

Reply:
UPDATE:
I've got compilation errors when I removed prefix from methods inside class `ScalaDocumentableContentBuilder`
Error:
```
[error] -- [E007] Type Mismatch Error: /home/fzybala/dotty-dokka/src/main/scala/dotty/dokka/ScalaPageCreator.scala:73:67 
[error] 73 |                        .group(kind = ContentKind.Symbol){ gbdr => gbdr
[error]    |                                                                   ^
[error]    |Found:    dotty.dokka.ScalaPageContentBuilder#ScalaDocumentableContentBuilder
[error]    |Required: Nothing
[error] 74 |                            .text("Companion ")
[error] 75 |                            .driLink(
[error] 76 |                                clazz.get(ClasslikeExtension).kind match {
[error] 77 |                                    case dotty.dokka.Kind.Object => "class"
[error] 78 |                                    case _ => "object"
[error] 79 |                                },
[error] 80 |                                co
[error] 81 |                            )
```

Reply:
That shouldn't happen, can you upload this version of code on a separate branch?

Reply:
https://github.com/lampepfl/scala3doc/tree/content-builder-with-error